File: leetcode/0219.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        auto n = nums.size();
        if(k == 0)
            return false;
        unordered_set s={nums[0]};
        for(int i = 1; i < n;i++){
            if(s.count(nums[i]))
                return true;
            s.emplace(nums[i]);
            if (i >= k)
                s.erase(nums[i-k]);
        }
        return false;
    }
};

File: leetcode/0219.go
package leetcode

func containsNearbyDuplicate(nums []int, k int) bool {
	s := map[int]bool{}
	for i, _ := range nums {
		if c := s[nums[i]]; c {
			return true
		}
		s[nums[i]] = true
		if i >= k {
			s[nums[i-k]] = false
		}
	}
	return false
}
